ngraph/core/src/type.cpp

```diff
@@ -4,20 +4,47 @@
 
 #include "ngraph/type.hpp"
 
-#include "ngraph/util.hpp"
+#include "openvino/util/common_util.hpp"
 
 namespace std {
 size_t std::hash<ngraph::DiscreteTypeInfo>::operator()(const ngraph::DiscreteTypeInfo& k) const {
-    NGRAPH_SUPPRESS_DEPRECATED_START
-    size_t name_hash = hash<string>()(string(k.name));
-    size_t version_hash = hash<decltype(k.version)>()(k.version);
-    // don't use parent for hash calculation, it is not a part of type (yet)
-    return ngraph::hash_combine(vector<size_t>{name_hash, version_hash});
-    NGRAPH_SUPPRESS_DEPRECATED_END
+    return k.hash();
 }
 }  // namespace std
 
 namespace ov {
+
+size_t DiscreteTypeInfo::hash() const {
+    if (hash_value != 0)
+        return hash_value;
+    size_t name_hash = name ? std::hash<std::string>()(std::string(name)) : 0;
+    size_t version_hash = std::hash<decltype(version)>()(version);
+    size_t version_id_hash = version_id ? std::hash<std::string>()(std::string(version_id)) : 0;
+
+    return ov::util::hash_combine(std::vector<size_t>{name_hash, version_hash, version_id_hash});
```

Comment:
why this value is not saved to `hash_value`?

Reply:
It is a constant method.

Reply:
`mutable` hash_value